source/code/include/playfab/PlayFabCallRequestContainer.h

```diff
@@ -24,13 +24,16 @@ namespace PlayFab
         std::string GetFullUrl() const;
         std::shared_ptr<PlayFabApiSettings> GetApiSettings() const;
         std::shared_ptr<PlayFabAuthenticationContext> GetContext() const;
+        std::string GetRequestId() const;
+        void SetRequestId(const std::string& newRequestId);
```

Comment:
is this the x-requestid header that's included in the api response?

Reply:
I suppose I'm asking because I'm wondering if bumblelion's custom http plugin should be interacting with this at all? is this used internally in the SDK anywhere that would require a custom plugin to have set this?

---
In reply to: [348113780](https://github.com/PlayFab/XPlatCoreTemplate/pull/47#discussion_r348113780) [](ancestors = 348113780)

Reply:
Yes, this is the PlayFab X-RequestID.
It's in the PlayFab specific CallRequestContainer, rather than the agnostic BaseCallRequestContainer, so this is the correct segregation.
As far as the HTTP layers go, yes, there's a lot of cross-contamination of PlayFab in those, and I agree it's a problem. But it's not anywhere near related to this PR.

Reply:
These changes are a consequence of the separate independent console repos, which had this change as a console-specific element, but we're moving certain interface pieces to core to reduce the complexity of the console-specific over-writes.